poetry.lock

```diff
@@ -0,0 +1,692 @@
+[[package]]
```

Comment:
NB: This lock file is only used for developers of this library. It is in no way included in the distribution - the distribution will use the floating requirements of `pyproject.toml`.

The lockfile is nice for development becuase it means transitive dependencies won't change on us overnight like has happened with `more-itertools` breaking `pytest` in Pants.

Reply:
If dependencies do change they should break though since they will in the wild. The convenience for developers is just a delay of the eventual break out in the real world.